### PR TITLE
Adding support for storing logits from zoo models

### PR DESCRIPTION
**Pull request by brimoor**
Adds a `SampleCollection.apply_model(..., store_logits=True)` argument that allows users to opt-in to populating the `logits` fields of their predictions.

Only the `Classification` and `Classifications` classes support logits, so `store_logits=True` is only allowed for classifiers.

All models in the zoo declare whether they support storing logits via a `Model.has_logits = True/False` property

Here are the models that now support logits:

```py
for model_name in foz.list_zoo_models():
    zoo_model = foz.get_zoo_model(model_name)
    if zoo_model.has_tag("logits"):
        print(model_name)
```

```
alexnet-imagenet-torch
densenet121-imagenet-torch
densenet161-imagenet-torch
densenet169-imagenet-torch
densenet201-imagenet-torch
googlenet-imagenet-torch
inception-resnet-v2-imagenet-tf1
inception-v3-imagenet-torch
inception-v4-imagenet-tf1
mnasnet0.5-imagenet-torch
mnasnet1.0-imagenet-torch
mobilenet-v2-imagenet-tf1
mobilenet-v2-imagenet-torch
resnet-v1-50-imagenet-tf1
resnet-v2-50-imagenet-tf1
resnet101-imagenet-torch
resnet152-imagenet-torch
resnet18-imagenet-torch
resnet34-imagenet-torch
resnet50-imagenet-torch
resnext101-32x8d-imagenet-torch
resnext50-32x4d-imagenet-torch
shufflenetv2-0.5x-imagenet-torch
shufflenetv2-1.0x-imagenet-torch
vgg11-bn-imagenet-torch
vgg11-imagenet-torch
vgg13-bn-imagenet-torch
vgg13-imagenet-torch
vgg16-bn-imagenet-torch
vgg16-imagenet-tf
vgg16-imagenet-torch
vgg19-bn-imagenet-torch
vgg19-imagenet-torch
wide-resnet101-2-imagenet-torch
wide-resnet50-2-imagenet-torch
```

Example usage:

```py
import fiftyone as fo
import fiftyone.zoo as foz

MODELS = [
    "inception-v3-imagenet-torch",
    "vgg16-imagenet-tf",
    "mobilenet-v2-imagenet-tf1",
]

dataset = foz.load_zoo_dataset("imagenet-sample", max_samples=5, shuffle=True)

label_fields = []
for model_name in MODELS:
    model = foz.load_zoo_model(model_name)
    label_field = model_name.lower().replace("-", "_").replace(".", "_")
    dataset.apply_model(
        model,
        label_field=label_field,
        confidence_thresh=None,  # ensure all samples have predictions
        store_logits=True,
    )
    label_fields.append(label_field)

# Check that logits exist
for sample in dataset:
    print(sample.filepath)
    for field in label_fields:
        print(sample[field].logits.shape)
```

### CODE REVIEW

**Review by ehofesmann — APPROVED** [review]
LGTM! It works great. 

I realized that we don't have any CIFAR10 models though so there is still some work to be done before this is able to be used in the blog post.
